File: Master_the_Key_ROS_Concepts/Lec1_Create_a_Cakin_Workspace/catkin_ws/src/my_robot_tutorials/src/smartphone.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

void callback_receive_radio_data(const std_msgs::String& msg)
{
	ROS_INFO("Message Received : %s", msg.data.c_str()); //c_str() is convert data to string
}

int main(int argc, char **argv)
{
	ros::init(argc, argv, "smartphone");
	ros::NodeHandle nh;

	// Create subscriber
	ros::Subscriber sub = nh.subscribe("/robot_news_radio", 1000, callback_receive_radio_data);

	ros::spin(); //keep node runing till we shutdown
}

File: Master_the_Key_ROS_Concepts/Lec1_Create_a_Cakin_Workspace/catkin_ws/src/my_robot_tutorials/scripts/smartphone.py
#!/usr/bin/env python3

import rospy
from std_msgs.msg import String

def callback_receive_radio_data(msg):
    rospy.loginfo("Message Received: ")
    rospy.loginfo(msg)


if __name__ == '__main__':
    rospy.init_node('smartphone')


    # Sbuscriber topic .. use same name of topic
    sub = rospy.Subscriber("/robot_news_radio",String, callback_receive_radio_data)
    
    #something that keep this scrip to have callback
    rospy.spin()
